### PR TITLE
Integrate 'DRNG test preparation' into OpenQA

### PR DESCRIPTION
**Pull request by Amrysliu**
Related: https://progress.opensuse.org/issues/108485
Verify run: https://openqa.suse.de/tests/8339847 (s390x)
                  https://openqa.suse.de/tests/8339699 (x86)
                  https://openqa.suse.de/tests/8340047 (aarch64)

### CODE REVIEW

File: tests/security/atsec/drng_test_preparation.pm [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    # Complile gather_random_data
+    my $exe_file = 'gather_random_data';
+    assert_script_run('cd /usr/local/atsec');
+    assert_script_run("gcc -o $exe_file -lcrypto -lssl -lgcrypt gather_random_data.c");
+
+    # Prepare the test directory
+    assert_script_run('mkdir -p /root/eval/drng');
+
+    # Copy the executable file to test directory
+    assert_script_run('cp gather_random_data /root/eval/drng/');
```
Comment (lilyeyes) [inline tests/security/atsec/drng_test_preparation.pm:33]: Why not use "$exe_file" here?

Comment (lilyeyes) [reply tests/security/atsec/drng_test_preparation.pm:33]: Please define a variable (e.g., $f_drmg) for '/root/eval/drng/'.

File: tests/security/atsec/drng_test_preparation.pm [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    select_console 'root-console';
+
+    # Install the required packages
+    zypper_call('in gcc libopenssl-devel libgcrypt-devel');
+
+    # Complile gather_random_data
+    my $exe_file = 'gather_random_data';
+    assert_script_run('cd /usr/local/atsec');
+    assert_script_run("gcc -o $exe_file -lcrypto -lssl -lgcrypt gather_random_data.c");
```
Comment (rfan1) [inline tests/security/atsec/drng_test_preparation.pm:27]: gather_random_data.c -> $exe_file.c

File: tests/security/atsec/drng_test_preparation.pm [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+    # Copy the executable file to test directory
+    assert_script_run('cp gather_random_data /root/eval/drng/');
+
+    # Run the test
+    assert_script_run('cd /root/eval/drng/');
+    assert_script_run("./$exe_file 2");
+
+    # Check result
+    my $result = script_output('ls -l | grep -v "total"');
```
Comment (rfan1) [inline tests/security/atsec/drng_test_preparation.pm:40]: @Amrysliu Can you please add some comments here?
1. why grep -v total?
2. expect result?

File: tests/security/atsec/drng_test_preparation.pm [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    my @lines = split(/\n/, $result);
+    record_info('ERROR', "Others files are generated\n$result", result => 'fail') if (scalar(@lines) != 4);
```
Comment (rfan1) [inline tests/security/atsec/drng_test_preparation.pm:42]: What is "Others" files?

File: tests/security/atsec/setup_atsec_env.pm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+sub run {
+    my ($self) = shift;
+
+    select_console 'root-console';
+
+    # Install tool packages
+    zypper_call('in wget');
+
+    # Download the test scripts
+    my $code_path = get_required_var('CODE_PATH');
```
Comment (rfan1) [inline tests/security/atsec/setup_atsec_env.pm:25]: @Amrysliu Can we put the scripts into ~data dir?

File: tests/security/atsec/drng_test_preparation.pm [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+    foreach my $line (@lines) {
+        my @items = split(/\s/, $line);
+        my $file_size = $items[4];
+        my $file_name = $items[-1];
+
+        # Skip gather_random_data
+        next if ($file_name eq $exe_file);
```
Comment (lilyeyes) [inline tests/security/atsec/drng_test_preparation.pm:50]: Just suggestion: we can use `grep -v "total" | grep -v $exe_file` in lines `# Check result` to handle these 2 lines.

**Review by lilyeyes — APPROVED** [review]

File: tests/security/atsec/setup_atsec_env.pm (hunk 5 above)
Comment (lilyeyes) [reply tests/security/atsec/setup_atsec_env.pm:25]: The code/data put in` "data/"` is opensource this `*.c` file is from ATSEC and is confidential. @rfan1

File: tests/security/atsec/drng_test_preparation.pm (hunk 4 above)
Comment (Amrysliu) [reply tests/security/atsec/drng_test_preparation.pm:42]: The files are not expected.

File: tests/security/atsec/drng_test_preparation.pm (hunk 2 above)
Comment (rfan1) [reply tests/security/atsec/drng_test_preparation.pm:28]: @Amrysliu did you miss my comment?

**Review by rfan1 — APPROVED** [review]
lgtm

**Pull request merged**